### DISCUSSION TITLE
'pyslim' has no attribute 'recapitate'

### DISCUSSION THREAD

**Discussion by KNLawrence01** [tskit-dev/pyslim#267] [Q&A] [answered]
Hey all, 
I am running a code to recapitate a SLiM tree in msprime, but I'm having errors with pyslim. Here is the code I'm using:

import msprime, pyslim
import tskit
ts = tskit.load("MS123sweepEx.tree")
rts = pyslim.recapitate(ts,
            recombination_rate=1e-8,
            ancestral_Ne=200, random_seed=5)

I keep getting the error:

AttributeError: module 'pyslim' has no attribute 'recapitate'

Is there something wrong with my code or is there a better way of recapitating?
Thanks!

**Comment by petrelharp** [comment, marked as answer]
Hi, @oxPSEUDOxo - I believe this means your pyslim installation is out of date - can you reinstall and/or check the version? (It should be at 0.700.)

**Reply by KNLawrence01** [reply]
Hey, @petrelharp - That was a problem and I've since upgraded it to 0.700, but now I'm getting a different error and I'm not sure what to make of it. It is a few pages long (here is what the beginning of the error looks like).
![Failrecap](https://user-images.githubusercontent.com/108096710/175660698-b41295ed-d16e-4f26-bd3d-63f917e299ca.JPG)

**Reply by petrelharp** [reply]
Hm- it looks to me like you still have not got pyslim v0.700. You can check with `print(pyslim.__version__)` - could you verify?

**Reply by KNLawrence01** [reply]
I checked version and its stating that I have v0.700. Its throwing this error when I input any tree created inlcuding examples. I've used the examples from the SLiM Manual on page 471 and from tskit, https://tskit.dev/pyslim/docs/latest/tutorial.html#recapitation .

**Reply by KNLawrence01** [reply]
If I'm interpeting the errors right then it appears to be an issue with metadata which I'm not overly familiar with. Would having a defined metadata correct these errors?

**Reply by petrelharp** [reply]
Okay, my guess is now that it's an incompatibility between the older version of pyslim (0.700) and the newer version of SLiM (v4). If you're using SLiM v4 then you should use the pyslim beta release for 1.0: https://pypi.org/project/pyslim/1.0b1/

If not, could you post the full code that produces the error, including how the tree sequence you're reading in is produced?
